internal/security/proxy/requestor.go

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package proxy
+
+import (
+	"crypto/tls"
+	"net/http"
+	"time"
+)
+
+type Requestor interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func NewRequestor(skipVerify bool) Requestor {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
+	return &http.Client{Timeout: 10 * time.Second, Transport: tr}
```

Comment:
It would be nice to parametrize the timeout in second as provision with default value so the caller can configure it if desired